sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -416,12 +418,27 @@ object LogicalPlanIntegrity {
   }
 
   def validateSchemaOutput(previousPlan: LogicalPlan, currentPlan: LogicalPlan): Option[String] = {
-    if (!DataTypeUtils.equalsIgnoreNullability(previousPlan.schema, currentPlan.schema)) {
-      Some(s"The plan output schema has changed from ${previousPlan.schema.sql} to " +
-        currentPlan.schema.sql + s". The previous plan: ${previousPlan.treeString}\nThe new " +
-        "plan:\n" + currentPlan.treeString)
+    val isValid = if (SqlApiConf.get.caseSensitiveAnalysis) {
+      DataTypeUtils.equalsIgnoreNullability(previousPlan.schema, currentPlan.schema)
     } else {
+      if (!DataTypeUtils.equalsIgnoreNullability(previousPlan.schema, currentPlan.schema)) {
+        logInfo("The plan output schema has changed from ${previousPlan.schema.sql} to " +
+          currentPlan.schema.sql + s". The previous plan: ${previousPlan.treeString}\nThe new " +
+          "plan:\n" + currentPlan.treeString)
+        val CaseNotPreservingProblem = new SparkException("CaseNotPreservingRule")
+        val stackTraceString = Option(CaseNotPreservingProblem)
+          .map(Throwables.getStackTraceAsString).getOrElse("")
+        logInfo(stackTraceString)
+      }
+      DataTypeUtils.equalsIgnoreCaseAndNullability(previousPlan.schema, currentPlan.schema)
+    }
+
+    if (isValid) {
       None
+    } else {
+      Some(s"The plan output schema has changed from ${previousPlan.schema.sql} to " +
```

Comment:
this string is duplicated with L425-427 above. Can you deduplicate into a common place like a `def` or `val`?